Function Attributes/README.md

```diff
@@ -0,0 +1,524 @@
+Such memory blocks are reclaimed again by the GC by an algorithm called garbage collection.
+
+Some commonly used D operations take advantage of the GC as well. For example, elements of arrays live on dynamic memory blocks:
+
+```d
+// A function that takes advantage of the GC indirectly
+int[] append(int[] slice)
+{
+    slice ~= 42;
+    return slice;
+}
+```
+
+If the slice does not have sufficient capacity, the `~=` operator above allocates a new memory block from the GC.
+
+Although the GC is a significant convenience for data structures and algorithms, memory allocation and garbage collection are costly operations that make the execution of some programs noticeably slow.
+
+`@nogc` means that a function cannot use the GC directly or indirectly:
+
+```d
+void foo() @nogc
+{
+    // ...
+}
+```
+
+The compiler guarantees that a `@nogc` function does not involve GC operations.
+For example, the following function cannot call `append()` because it does not provide the @nogc guarantee:
+```d
+void foo() @nogc
+{
+    int[] slice;
+    // ...
+    append(slice); // compilation ERROR - @nogc function 'foo' cannot call non-@nogc function 'append'
+}
+```
+
+For an extensive list of operations forbidden in `@nogc` code, check this [link](https://dlang.org/spec/function.html#nogc-functions)
+
+## Code safety attributes
+
+As you might already imagine, `@safe`, `@trusted`, and `@system` are attributes that say something about the code safety that a function provides.
+As with purity, the compiler infers the safety level of templates, delegates, anonymous functions, and `auto` functions.
+
+### `@safe`
+
+A `@safe` function is a function that is statically verified by the compiler to ensure that there are no operations that might lead to memory corruption.
+In a `@safe` function, there are several language features that cannot be used, such as casts that break the type system or pointer arithmetic.
+A full list of operations that are forbidden in `@safe` functions can be found [here](https://dlang.org/spec/function.html#safe-functions).
+In spite of these limitations, these functions provide strong safety guarantees that are very important.
+
+Let's take a look at this unsafe code from below:
+
+```d
+import std.stdio : writeln;
+
+void main()
+{
+    int a = 7;
+    int b = 9;
+
+    /* some code later */
+
+    *(&a + 1) = 2;
+    writeln(b);
+    writeln(b);
+}
+```
+
+Running this code yields the result:
+
+```
+9
+2
+```
+
+Wait, what?
+No, this is not a typo: the value of b changes between two instructions for no apparent reason.
+This is happening because the compiler does not offer any guarantees when it comes to pointer arithmetic on local variables.
+Pointer arithmetic definition is by definition an unsafe operation and if the faulting line would have been hidden between another 1K lines of code, it would have taken a lot of time to get to the root of the problem.
```

Comment:
```suggestion
Pointer arithmetic is by definition an unsafe operation and if the faulting line would have been hidden between another 1K lines of code, it would have taken a lot of time to get to the root of the problem.
```